### PR TITLE
Steep track pieces for log flumes

### PR DESCRIPTION
**Pull request by karst**
I was thinking for a little while how to add these as this is a feature people have been looking for for quite a bit so I decided to get a start on this make the first few sprites and then pressure myself to make more pieces by having a PR open for it. 

This is basically the idea.
image
ᶜᵒⁿᶜᵉᵖᵗ ᵃʳᵗ

To get this working I am photoshopping the Splash Boats flume to fit retrofit on the log flume. 

To Do List
----------
- [x] Make pieces for SW-NE (Can I get the other set working by just mirroring them?)
- [x] Make pieces for NW-SE
- [x] Make pieces for NE-SW 
- [x] Make pieces for SE-NW
- [x] Fix shading and pixel errors
- [x] Get pieces working in game
- [x] Check bounding boxes & supports
- [x] Add pieces to extra track pieces in meta file (unless someone manages to update the log flume's sprites to fit on 60 degree 
- [x] Update Network Base
- [x] Add changelog entry

If I am forgetting anything let me know, if you want to help me out with any of this tag me on discord (especially if someone would like to see if they can add more sprites to the vehicles).

### CODE REVIEW

**Comment by karst** [comment]
Vanilla g1 palette isn't working correctly in g2. I believe #16538 is gonna be necessary to get the sprites to match perfectly.

**Comment by spacek531** [comment]
If you save the sprites as 8-bit using that branch it isn't necessary to be merged, simply annoying

**Comment by Gymnasiast** [comment]
Any news on this?

**Comment by karst** [comment]
Yea i wanted to pick this up soon, i need to recover from jetlag first tho :)

**Comment by karst** [comment]
Okay, I have got everything functional in game now.
image

Enough work for today, up next are bounding boxes, supports, pixel fixes and shading fixes.

**Comment by karst** [comment]
After hours of testing I have finally completed the first permanent piece. The rest should be straight forward after this but I am gonna have to recreate the from scratch. 
https://i.gyazo.com/92322a0d70dfc8fa4ca3c430a0f1c537.gif

In the end I have completed the edit in about 5 minutes so it's gonna take me a day off sometime to get all these textures to work and look properly.

**Comment by karst** [comment]
There of the four angles are now complete. I have tried fixing shading, as there's a lot of ugly vanilla shading, but this definitely looks a lot better.
**PICS!!!**
NWSE Front

NESW Front

NESW Back

Please ignore the supports! I haven't worked at all on these yet.

**Comment by karst** [comment]
Getting near to finishing, all that's left to do is bounding boxes.

**Comment by karst** [comment]
Sadly the supports on the back do not correctly, I left it like this if someone wants to fix that in the future (Rik perhaps). It's a bug that also occurs on something like the Dinghy slide. 
image

Let me know if there's anything I missed that needs fixing.

**Comment by 733737** [comment]
will the custom log be added to the game, that way we could enable steep hills without cheats...

**Comment by karst** [comment]
image
image
image
image
@Basssiiie  I have made the flume 2 px smaller at the back as requested. I have also fixed some more shading/line up issues. And no I have not been able to adjust the angle for obvious reasons 😬

File: src/openrct2/ride/water/meta/LogFlume.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ constexpr const RideTypeDescriptor LogFlumeRTD =
     SET_FIELD(NameConvention, { RideComponentType::Boat, RideComponentType::Track, RideComponentType::Station }),
     SET_FIELD(EnumName, nameof(RIDE_TYPE_LOG_FLUME)),
     SET_FIELD(AvailableBreakdowns, (1 << BREAKDOWN_SAFETY_CUT_OUT) | (1 << BREAKDOWN_CONTROL_FAILURE)),
-    SET_FIELD(Heights, { 9, 24, 7, 9, }),
+    SET_FIELD(Heights, { 10, 24, 7, 9, }),
```
Comment (Gymnasiast) [inline src/openrct2/ride/water/meta/LogFlume.h:40]: Why was this changed?

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
There are a few stray pixels here.
afbeelding

The far side has orange colouring here on the steep sections, even though the rest of it does not:
afbeelding

To a lesser extent, this also happens from another viewpoint:
afbeelding

And we’ll have to find some sort of solution for this (I know it’s not technically caused by this PR):
afbeelding

**Comment by karst** [comment]
I have addressed the top three, including this ugly mess
image
(I sent that one to the fitness so it could get a bit more muscle)

I think the bottom one should be addressed another time in a seperate PR. I will check if there's been a bug report for it and if not report it.

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
Still a few stray pixels here:

afbeelding

**Comment by Basssiiie** [comment]
This blue seems off as well, if it hasn't been fixed already.

image

**Comment by karst** [comment]
@Basssiiie that now looks like this 
image
Which stil isn't perfectd but it's miles better than what it was.
I will now work on that singular pixel Michael pointed out

**Comment by karst** [comment]
Anything else that needs patching?
image
image
image
image

**Review by Basssiiie — APPROVED** [review]

**Comment by Gymnasiast** [comment]
There was a second pixel in my screenshot that still has to be addressed. It’s not quite as easy to spot, though:

afbeelding

**Comment by karst** [comment]
OH THERE. Damn that blends in with grass and a half

**Comment by karst** [comment]
My 30 seconds are up

**Comment by karst** [comment]
@rik-smeets There's two cases where there's slight glitching with vehicles. I cannot seem to patch these myself. 
One is at the bottom (60-25) on the back (nwse). I feel this issue is because the sprite is too thin, not because of bounding boxes. The other is at the front (60 down) at nwse, but it seems like the bounding boxes of this direction are broken in general.

Lastly there's a glitch with supports displaying behind the track rather than on top of it which I cant seem to fix either.

Other than that this PR should be good to go and these fixes might be a better fit for a different general fix.

**Comment by karst** [comment]
I think I could patch the issue with the back NWSE by making the butt a pixel thicker. 
Other two issues are reported bugs #18613 and #18630, which have nothing to do with this PR sadly.

**Comment by karst** [comment]
I suppose since there's no official vehicle out for the flume yet it might be smarter to look into that poking out with the official model instead? Something Realsteel could look into perhaps?

**Comment by Realsteel89** [comment]
> 

this boat was my very first attempt and i made a big mistake. i assumed that the line has a water volume which it doesn't have and i didn't implement the boat on the line like in the picture but underneath. This is also the reason why it glitches through the track (blue line).

98ed3c77ef9a89a6442abf8c99b32405
f0fb233ad584963cd1a2b162e0a60956

now i did an better version with the new renderer from x71 and it looked correct on the track
image

image
rn_image_picker_lib_temp_43d49afe-ed4b-45fa-922c-cea2e44c7b82
rn_image_picker_lib_temp_a7e9d2c1-44e8-450c-9620-89d80d51aef2

And yes i had the track version only with the small glitch on the track but I think this has already been fixed.

I can't release the new boot because the turntable doesn't work yet and object versioning doesn't exist. I could only release it for testing but then you would have to delete it afterwards.

**Comment by karst** [comment]
Wouldn't it be smart to up the save version by one so someone in a future pr or this PR could implement previous parks to have the sprites render invisible? I think the steep pieces on log flumes is one of the most used things that doesn't have sprites for it.
If it's below the new version when importing a park the sprites would be made invisible with the tile inspector automatically, while the new version and up wouldn't do that.

File: src/openrct2/park/ParkFile.cpp [hunk 2]
```diff
@@ -1018,6 +1018,19 @@ namespace OpenRCT2
                                         }
                                     }
                                 }
+                                // Lots of Log Flumes exist where the downward slopes are simulated by using other track
```
Comment (Gymnasiast) [inline src/openrct2/park/ParkFile.cpp:1021]: Note to self: remove this text, it’s been moved to the function in question.

**Review by Gymnasiast — APPROVED** [review]
LGTM, asking @rik-smeets to take it for a test.

**Comment by rik-smeets** [comment]
From this angle, there seems to be a slight mismatch in the transition between the steep and sloped track pieces:
image

This isn't visible from the other angle:
image

Could you see if you could fix that, @karst? Other than that, LGTM. Very nice addition!

**Comment by Gymnasiast** [comment]
@karst Do not commit or push anything to the branch until you’ve checked with me, otherwise it’s going to get very messy ;)

**Comment by karst** [comment]
Ah, i did already try slope it better than originally, but I'll look into it one more time then. Expect tomorrow or the day after

**Comment by karst** [comment]
image
I tried my hardest improving the transition. I hope this is good enough

**Comment by rik-smeets** [comment]
Looks good to me @karst, thanks! Will you coordinate with @Gymnasiast to add it to this PR?

**Review by rik-smeets — APPROVED** [review]

**Comment by Legotown55** [comment]
When will this be murged it to the game

**Pull request merged**